radix-engine/src/engine/call_frame.rs

```diff
@@ -261,29 +266,12 @@ impl<'a> LoadedSNodeState<'a> {
                 BorrowedSNodeState::Resource(resource_address, resource_manager) => {
                     track.return_borrowed_global_mut_value(resource_address, resource_manager);
                 }
-                BorrowedSNodeState::Bucket(_bucket_id, _bucket) => {}
-                BorrowedSNodeState::Proof(_proof_id, _proof) => {}
-                BorrowedSNodeState::Vault(vault_id, vault, value_type) => match value_type {
-                    ValueType::Owned => {
-                        owned_values.insert(
-                            StoredValueId::VaultId(vault_id.clone()),
-                            StoredValue::Vault(vault_id, vault),
-                        );
-                    }
-                    ValueType::Ref(ValueRefType::Uncommitted { root, ancestors }) => {
-                        let store = owned_values
-                            .get_mut(&StoredValueId::KeyValueStoreId(root))
-                            .map(|v| match v {
-                                StoredValue::KeyValueStore(_, store) => store,
-                                _ => panic!("Expected KV store"),
-                            })
-                            .unwrap();
-                        store.put_child_vault(&ancestors, vault_id, vault);
-                    }
+                BorrowedSNodeState::TrackedVault(vault_id, vault, value_type) => match value_type {
                     ValueType::Ref(ValueRefType::Committed { component_address }) => {
                         track
                             .return_borrowed_global_mut_value((component_address, vault_id), vault);
                     }
+                    _ => panic!("Should not get here."),
```

Comment:
Minor: in the long run, we should be careful with documenting our assumption. Instead of saying it's not reachable, we should say why it's not reachable and enforced by what. For this specific case, I would document "Tracked vaults are owned by Track and only references are passed to call frames" although it's quite obvious.